src/manifest.rs

```diff
@@ -395,10 +406,18 @@ impl LocalManifest {
     pub fn upgrade(&mut self, dependency: &Dependency, dry_run: bool) -> Result<()> {
         for (table_path, table) in self.get_sections() {
             let table_like = table.as_table_like().expect("Unexpected non-table");
-            for (name, _old_value) in table_like.iter() {
-                if name == dependency.name {
-                    self.manifest
-                        .update_table_entry(&table_path, dependency, dry_run)?;
+            for (name, toml_item) in table_like.iter() {
+                let dep_name = toml_item
+                    .as_inline_table()
```

Comment:
we should handle non-inline dependencies with renames as well
toml_edit has https://docs.rs/toml_edit/0.1.5/toml_edit/enum.Item.html#method.as_table_like to abstract over tables